testsuite/cluster-test/src/experiments/cpu_flamegraph.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::experiments::ExperimentParam;
+use crate::{
+    cluster::Cluster, experiments::Context, experiments::Experiment, instance, instance::Instance,
+};
+use futures::future::{BoxFuture, FutureExt};
+
+use crate::effects::{Action, GenerateCpuFlamegraph};
+use std::{
+    collections::HashSet,
+    fmt::{Display, Error, Formatter},
+    time::Duration,
+};
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+pub struct CpuFlamegraphParams {
+    #[structopt(
+        long,
+        default_value = "60",
+        help = "Number of seconds for which perf should be run"
+    )]
+    pub duration_secs: usize,
+}
+
+pub struct CpuFlamegraph {
+    duration_secs: usize,
+    perf_instance: Instance,
+}
+
+impl ExperimentParam for CpuFlamegraphParams {
+    type E = CpuFlamegraph;
+    fn build(self, cluster: &Cluster) -> Self::E {
+        let (perf_instance, _) = cluster.split_n_validators_random(1);
```

Comment:
Should we just make `random_validator` function in Cluster? Probably going to be used a lot.

Reply:
It already exists :-/